server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -145,15 +147,27 @@ onConn (L.Logger logger) corsPolicy wsId requestHead = do
       sendMsg wsConn SMConnKeepAlive
       threadDelay $ 5 * 1000 * 1000
 
+    jwtExpiryHandler wsConn = do
+      currTime <- TC.getCurrentTime
```

Comment:
This has to be done after the stm action. The STM action will return only after the connection is initialised so you'll sleep till after the expiry. Hence you need to do this after you get the expiry time.